scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1218,6 +1225,73 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 	return false;
 }
 
+/* Mapping of cpu to most delayed DSQ it can consume */
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__type(key, u32);
+	__type(value, u64);
+	__uint(max_entries, 1);
+} antistall_cpu_dsq SEC(".maps");
+
+/* Mapping cpu to delay of highest delayed DSQ it can consume */
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__type(key, u32);
+	__type(value, u64);
+	__uint(max_entries, 1);
+} antistall_cpu_max_delay SEC(".maps");
+
+bool antistall_consume(s32 cpu, struct cpu_ctx *cctx)
+{
+	u64 *antistall_dsq, *antistall_cpu_max, 
+			empty, runnable_at_sec, now_sec, cur_delay;
+	u32 zero;
+	bool consumed;
+	struct task_struct *p;
+	
+	cur_delay = 0;	
+	consumed = false;
+	zero = 0;
+	empty = -1;
```

Comment:
Note that while `(u64)-1` is an invalid DSQ ID, it might not necessarily stay that way. Please use `SCX_DSQ_INVALID`.